### ISSUE TITLE
Fix S3981 FN: Not raised for IList and ICollection

### ISSUE THREAD

**Issue by donnytian** [SonarSource/sonar-dotnet#3929]
### Description

Rule S3981: Not raised for IList and ICollection

### Repro steps

Please provide a code snippet that compiles which reproduces the problem. Alternatively, a link to a SonarCloud project where the issue appears, or a git repository, or attach a zip with a reproducer project.
```cs
private void S3981(IList<int> list, ICollection<int> collection)
{
    if (list.Count >= 0) { }

    if (collection.Count >= 0) { }
}
```
### Expected behavior

expect S3981 on line 3 and 5 for `IList` and `ICollection`

### Actual behavior

no issue reported
### Known workarounds

n/a
### Related information

* C#/VB.NET Plugins version - 8.16.0.25740
* Visual Studio version - VS2019
* MSBuild / dotnet version - netcore 3.1
* SonarScanner for .NET version (if used)
* Operating System

**Comment by Corniel** [comment]
Note that I you implement an `IList` or `ICollection` that returns a negative value for `Count` (in some cases).

**Comment by Corniel** [comment]
And Also Consider `IReadOnlyList`, `IReadOnlyCollection`, and others...

**Comment by pavel-mikula-sonarsource** [comment]
Hi @donnytian, thank you for reporting this case. I can confirm it as False Negative. 

Also thank you @Corniel for catching the `IReadOnlyCollection`. `IReadOnlyList` inherits `IReadOnlyCollection` so that one should be handled afterwards.

**Comment by pavel-mikula-sonarsource** [comment]
ToDo: 
* Issue is raised for types implementing the interface, but it's not raised for the interface itself.
* Add `IReadOnlyCollection`
